### ISSUE TITLE
[openhabcloud] Issue with item exposure in #1525

### ISSUE THREAD

**Issue by nesys** [openhab/openhab2-addons#4953]
After upgrading from #1502 to #1515 and after, the item exposure seems buggy.

In facts:

- the connection with openhab cloud is OK (instance is online, and in https://myopenhab.org/events I see correct data
- if I use Amazon Echo Binding, the items tagged are correctly exposed
- if I try to use items exposed via PaperUI, or via text file, the items in https://myopenhab.org/items are not changed accordingly. That was working in #1502, but now is not working (even #1525)
- https://myopenhab.org/notifications is reporting correctly

## Expected Behavior
switch item changed via classic UI should be also changed in https://myopenhab.org/items

## Current Behavior

Nothing happens. Also if I delete all items and events, events are recreated correctly, items not (after the requested action to see the item in myopenhab.org). I've also canceled and re-created from scratch my myopenhab account ... same issue

log:
2019-02-21 09:06:22.093 [WARN ] [io.openhabcloud.internal.CloudClient] - Received command from openHAB Cloud for item 'Andrea_Presence_IFTTT', which is not exposed.

So the recipe in IFTTT is working as expected. The issue is on OH side.

## Possible Solution
That is working in #1502. Maybe some changes in between (but before #1515) need to be verified

## Steps to Reproduce (for Bugs)
1. delete all items and events
2. expose an item via PaperUI
3. see in myopenhab.org -> nothing

## Context
Trying to work with exposed items and IFTTT recipes

## Your Environment
My current configuration is openhabian #1525 on vmware VM.

**Comment by nesys** [comment]
here the post in our community:
https://community.openhab.org/t/issue-deleting-items-and-events-no-back-online-item-exposure-bug/66754

**Comment by digitaldan** [comment]
Hi @nesys I'm trying to understand the issue

 you mention tagging which does not have any affect with myopenhab, the only items that will report state to myopenhab will be set in `paper ui  -> services -> openhab Cloud -Confiure -> Items to expose ` as in the picture below
![image](https://user-images.githubusercontent.com/1903737/53289575-7155c780-374c-11e9-9c9e-71b69ca5664e.png)

Is this what you are referring to?  Note that this is only really useful for IFTTT and does not have anything todo with tagging or exposing items to Alexa or Google home

>After upgrading from #1502 to #1515 and after, the item exposure seems buggy.

I am trying to understand what those 2 PR's have to do with this issue?  I just tried on my system and things seem to be working, but i'm not sure if you are running something different?

**Comment by wborn** [comment]
> I am trying to understand what those 2 PR's have to do with this issue?

@nesys is referring to openHAB 2.5.0-SNAPSHOT build numbers. :wink:

**Comment by nesys** [comment]
> Is this what you are referring to? Note that this is only really useful for IFTTT and does not have anything todo with tagging or exposing items to Alexa or Google home

Exactly. And of course I'm referring to use IFTTT with OH. But Alexa is working correctly. I know this is not related to expose items ... I was thinking that was a useful information for you in case. Also the log above seems the flow between IFTTT and OH Cloud is working correctly, the issue is between my OH and myopenhab.org

**Comment by digitaldan** [comment]
Got it , i understand now.  I'll take a look, i ended up downgrading until the UI migrations are finished, I'll upgrade my instance a take a look.

**Comment by nesys** [comment]
thanks for your support @digitaldan ... in facts from OH #1502 I see this issue (still in current OH #1534)

**Comment by digitaldan** [comment]
Hmm, so the cloud binding is no loner able to get the stored configuration from the OH core as it used to.  The config is empty and this line return a null/empty object:
https://github.com/openhab/openhab2-addons/blob/master/addons/io/org.openhab.io.openhabcloud/src/main/java/org/openhab/io/openhabcloud/internal/CloudService.java#L189

even though my config shows differently
```
Pid:            org.openhab.openhabcloud
BundleLocation: null
Properties:
   baseURL = https://myopenhab.org/
   expose = [F1_DansOffice_Light, ZONE4_STATUS, CloudZone1]
   mode = remote
   service.pid = org.openhab.openhabcloud
```

@wborn any ideas on what might have changed in the core?

**Comment by digitaldan** [comment]
@wborn Could this be related to this commit? It looks ok to me, but it does seem like the binding can not find its config service

https://github.com/openhab/openhab2-addons/commit/2ef23f032174dae89e67bbe2a9cc6063b3d29d86

**Comment by wborn** [comment]
What might help is adding `configurationPid = "org.openhab.openhabcloud"` to:

https://github.com/openhab/openhab2-addons/blob/99c346c199d2d6bf029b847ad462f6dce57e1ed8/addons/io/org.openhab.io.openhabcloud/src/main/java/org/openhab/io/openhabcloud/internal/CloudService.java#L70-L74

**Comment by digitaldan** [comment]
let me give it a try

**Comment by digitaldan** [comment]
👍  Yep, that did it, PR coming....

**Comment by nesys** [comment]
I can confirm, issue solved in  OH #1536

**Comment by digitaldan** [comment]
👍
